### ISSUE TITLE
`SIM115 Use context handler` across a try/else block

### ISSUE THREAD

**Issue by smurfix** [astral-sh/ruff#8221]
```
        try:
            f = open(f"moat.{attr}")
        except OSError:
            pass
        else:
            with f:
                do_something_complicated(f)
```
This complains `SIM115 Use context handler for opening files`, except that I *am* doing that – just not on the same line, because I don't want the exception handler to cover the `do_something_complicated` part.

It'd be nice if ruff recognized this pattern.

**Comment by smurfix** [comment]
This also happens with "normal" arguments.

**Comment by bartfeenstra** [comment]
I am experiencing similar issues, when using `open()` with an `ExitStack()`. It seems that the rule raises violations in cases where no `with` statement is detected, which is only one of the ways to enter and exit a context manager.
